### PR TITLE
feat(VERSION): Removed VERSION from perl modules

### PR DESCRIPTION
**Pull request by henrikstranneheim**
- Removed test_standard_cli sub

### This PR adds | fixes:

- See above
- Added exclusion of VERSION in each pm module from Perl critic

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [ ] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: t/mip_analyse_dragen_rd_dna.test [hunk 1]
```diff
@@ -61,7 +59,7 @@ GetOptions(
     ## Display version number
     q{v|version} => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION . $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
         exit;
```
Comment (jemten) [inline t/mip_analyse_dragen_rd_dna.test:63]: This option has no function anymore and should probably removed.

File: t/mip_analyse_rd_dna.test [hunk 2]
```diff
@@ -76,7 +74,7 @@ GetOptions(
     ## Display version number
     q{v|version} => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION . $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
         exit;
```
Comment (jemten) [inline t/mip_analyse_rd_dna.test:78]: Same as above. And goes for all .test files

Comment (jemten) [reply t/mip_analyse_rd_dna.test:78]: Looked a bit further and the install.test files uses MIP_VERSION. Maybe these should to?

File: t/processes_add_job_id_dependency_tree.t [hunk 3]
```diff
@@ -43,7 +42,7 @@ GetOptions(
     },    #Display help text
     'v|version' => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION, $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
```
Comment (jemten) [inline t/processes_add_job_id_dependency_tree.t:45]: delete this and one more version further down

File: t/processes_add_parallel_chains_job_ids_to_job_id_dependency_string.t [hunk 4]
```diff
@@ -43,7 +40,7 @@ GetOptions(
     },    #Display help text
     'v|version' => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION, $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
```
Comment (jemten) [inline t/processes_add_parallel_chains_job_ids_to_job_id_dependency_string.t:43]: delete version here and further down

File: t/processes_add_parallel_job_id_to_case_id_dependency_tree.t [hunk 5]
```diff
@@ -43,7 +40,7 @@ GetOptions(
     },    #Display help text
     'v|version' => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION, $NEWLINE;
```
Comment (jemten) [inline t/processes_add_parallel_job_id_to_case_id_dependency_tree.t:46]: version

File: t/processes_add_parallel_job_id_to_parallel_dependency_tree.t [hunk 6]
```diff
@@ -43,7 +40,7 @@ GetOptions(
     },    #Display help text
     'v|version' => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION, $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
```
Comment (jemten) [inline t/processes_add_parallel_job_id_to_parallel_dependency_tree.t:43]: delete this

File: t/processes_add_parallel_job_ids_to_job_id_dependency_string.t [hunk 7]
```diff
@@ -43,7 +40,7 @@ GetOptions(
     },    #Display help text
     'v|version' => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION, $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
```
Comment (jemten) [inline t/processes_add_parallel_job_ids_to_job_id_dependency_string.t:43]: delete this  and version further down

File: t/create_log4perl_config.t [hunk 8]
```diff
@@ -47,7 +46,7 @@ GetOptions(
     # Display version number
     q{v|version} => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION . $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
```
Comment (jemten) [inline t/create_log4perl_config.t:49]: is this meaningful

File: t/get_path_entries.t [hunk 9]
```diff
@@ -46,7 +45,7 @@ GetOptions(
     # Display version number
     q{v|version} => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION . $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
```
Comment (jemten) [inline t/get_path_entries.t:48]: delete

File: t/gnu_chmod.t [hunk 10]
```diff
@@ -40,7 +39,7 @@ GetOptions(
     },    #Display help text
     q{v|version} => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE, basename($PROGRAM_NAME), $SPACE, $VERSION, $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
```
Comment (jemten) [inline t/gnu_chmod.t:42]: delete

File: t/gnu_ln.t [hunk 11]
```diff
@@ -40,7 +39,7 @@ GetOptions(
     },    #Display help text
     q{v|version} => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE, basename($PROGRAM_NAME), $SPACE, $VERSION, $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
```
Comment (jemten) [inline t/gnu_ln.t:42]: delete

Comment (jemten) [reply t/gnu_ln.t:42]: also a version further down

File: t/gnu_make.t [hunk 12]
```diff
@@ -45,7 +44,7 @@ GetOptions(
     # Display version number
     q{v|version} => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION . $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
```
Comment (jemten) [inline t/gnu_make.t:47]: delete

File: t/initiate_logger.t [hunk 13]
```diff
@@ -47,7 +46,7 @@ GetOptions(
     # Display version number
     q{v|version} => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION . $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
```
Comment (jemten) [inline t/initiate_logger.t:49]: delete

File: t/mip_core.t [hunk 14]
```diff
@@ -101,7 +99,7 @@ GetOptions(
     # Display version number
     q{v|version} => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION . $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
```
Comment (jemten) [inline t/mip_core.t:102]: delete

File: t/processes_add_pan_job_id_to_sample_id_dependency_tree.t [hunk 15]
```diff
@@ -43,7 +40,7 @@ GetOptions(
     },    #Display help text
     'v|version' => sub {
         done_testing();
-        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $VERSION, $NEWLINE;
+        say {*STDOUT} $NEWLINE . basename($PROGRAM_NAME) . $SPACE . $NEWLINE;
```
Comment (jemten) [inline t/processes_add_pan_job_id_to_sample_id_dependency_tree.t:43]: delete

**Review by jemten — APPROVED** [review]
Great regexp skills! 🙇 

The .test files still have the option version but does not print a version. Maybe all these files should use `$MIP_VERSION`

Looks lie you missed this in some tests: `$MIP::Processmanagement::Processes::VERSION, Perl $^V, $EXECUTABLE_NAME"`  
Also `version` is a part of quite a few `GetOptions` in the tests. I started commenting but there might actually be too many to do something about it right now. We could leave them in  and take them one by one as we update subs later.

File: t/mip_analyse_rd_dna.test (hunk 2 above)
Comment (henrikstranneheim) [reply t/mip_analyse_rd_dna.test:78]: I switched them to use $MIP_VERSION

File: t/create_log4perl_config.t (hunk 8 above)
Comment (henrikstranneheim) [reply t/create_log4perl_config.t:49]: Fixed

File: t/get_path_entries.t (hunk 9 above)
Comment (henrikstranneheim) [reply t/get_path_entries.t:48]: Fixed

**Comment by henrikstranneheim** [comment]
Ok, I did the ones above. I think any other ones will have to be done as we touch the code in the future

**Comment by jemten** [comment]
> Ok, I did the ones above. I think any other ones will have to be done as we touch the code in the future

Yeah. It's too much otherwise

**Pull request merged**